### PR TITLE
Yixiao summarybar unittest

### PR DESCRIPTION
**Pull request by JYXiao-2021**
# Description
Create the unit test for the SummaryBar component to confirm whether it working as intended.

## Main changes explained:
- Create file SummarBar.test.js to do the unit test.

## How to test:
1.check into current branch
2.do npm install to run this PR locally
3.make sure the module all download
3.run npm test SummaryBar.test.js

## Screenshots or videos of changes:
image

### CODE REVIEW

**Review by Alakabtaha0 — APPROVED** [review]
Correctly working when running test.

**Review by aaryaneil — APPROVED** [review]
Tested your PR and all test cases passed.
Screenshot 2024-04-20 at 5 17 21 PM

**Review by alla0810 — APPROVED** [review]
All tests are passing.  Nice work!

image

**Review by jennZhang93 — APPROVED** [review]
Tests all passed locally. Good work!
image

**Comment by jennZhang93** [comment]
I am still learning unit testing, and I do have a question about the code here. After line 62, shouldn't there be an `expect` statement testing if the component was actually changed?

**Review by curiousfancier — APPROVED** [review]
I run the all test cases and they have successfully passed

image

File: src/components/SummaryBar/__tests__/SummaryBar.test.js [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+
+  it('should display loading when displayUserProfile and summaryBarData are undefined', () => {
+    wrapper.setProps({ displayUserProfile: undefined, summaryBarData: undefined });
+    expect(wrapper.text()).toEqual('');
+  });
+  
+
+  it('updates correctly when props change', () => {
+    wrapper.setProps({ displayUserId: '456' }); 
+
```
Comment (Parth-tech) [inline src/components/SummaryBar/__tests__/SummaryBar.test.js:63]: I had a question is there a way we can check on this line wether or not was updated with the props correctly?

**Review by Parth-tech — APPROVED** [review]
Hey, 
All the test cases passed successfully. Just had a small question for which I have left a comment. Thanks!
Screenshot 2024-04-27 at 3 53 30 PM

**Review by Tiantian-C — APPROVED** [review]
All the tests have been passed!
Screenshot 2024-04-27 at 5 25 10 PM

**Comment by beblicarl** [comment]
All test cases passed successfully

image

**Review by beblicarl — APPROVED** [review]

**Comment by one-community** [comment]
Thank you all, merging!

**Pull request merged**